### PR TITLE
upgrade django oscar to latest version

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
LEARNER-2634

FYI: @edx/learner @edx/learner-spartans

### CODE REVIEW

**Comment by ahsan-ul-haq** [comment]
@mikedikan @mjfrey please review

**Comment by mikedikan** [comment]
@ahsan-ul-haq - I just connected with @mjfrey and we will be able to review, but not today.

**Comment by ahsan-ul-haq** [comment]
@mikedikan thats great thanks for the follow up

**Comment by ahsan-ul-haq** [comment]
Here are number of records in in the tables which will be affected by migrations
```
catalogue_productattributevalue = 38255
catalogue_productattribute = 13
customer_email = 254
offer_conditionaloffer = 18916
voucher_voucher = 174274
voucher_voucherapplication = 63398
```

**Review by mjfrey — APPROVED** [review]
Did some local testing and seems fine.  Please test this a lot on stage before releasing. Also give devops a heads up about the migrations.

File: ecommerce/extensions/offer/models.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def clean_value(self):
                 'Failed to create Benefit. Benefit value may not be a negative number.'
             )
 
+    def clean_percentage(self):
```
Comment (bderusha) [inline ecommerce/extensions/offer/models.py:40]: How does this get called?

**Review by bderusha — APPROVED** [review]
In general when we see `AlterField` in migrations we need to be cautious.  For large tables this can cause an issue with our deployments.  In this case the tables in question are small so it should be fine to have a lock on that table for a few milliseconds while the changes are applied.

File: ecommerce/extensions/offer/models.py (hunk 1 above)
Comment (ahsan-ul-haq) [reply ecommerce/extensions/offer/models.py:40]: This get called when `Benefit` object is created.

**Pull request merged**